Source/ACE.Database/ShardDatabase.cs

```diff
@@ -592,23 +602,31 @@ public bool SaveObject(AceObject aceObject)
             SaveObjectInternal(transaction, aceObject);
 
             // Do we have any inventory to save - if not, we are done here?
-            if (aceObject.Inventory.Count <= 0)
-                return transaction.Commit().Result;
-
-            foreach (AceObject invItem in aceObject.Inventory.Values)
+            if (aceObject.Inventory.Count > 0)
             {
-                SaveObjectInternal(transaction, invItem);
+                foreach (AceObject invItem in aceObject.Inventory.Values)
```

Comment:
Sent you a note about this via Discord.

Reply:
reworked - I think the problem might be in a different area.